packages/swayze/lib/src/core/viewport_context/viewport_context.dart

```diff
@@ -162,6 +167,22 @@ class ViewportAxisContextState {
   /// Just like [visibleIndices] but for the [frozenRange]
   final Iterable<int> visibleFrozenIndices;
 
+  /// Wetter or not the header is being dragged.
+  final bool isDragging;
+
+  /// Headers that are being dragged.
+  final Range draggingHeaders;
+
+  /// Current dragging reference, eg, the current header that [draggingPosition]
+  /// is hovering.
+  final int draggingCurrentReference;
+
+  /// Current dragging position.
+  final Offset draggingPosition;
+
+  /// Extent of all headers being dragged.
+  final double draggingHeaderExtent;
```

Comment:
Should this be moved into a class?

Reply:
Maybe, I was thinking about that. Since all other properties are added to the context without a class, I kept the same standard. But I think we are at a point where we have too many properties here...

Reply:
I agree it should be moved into its own class. Although all these properties have default values, they don't really make sense when not dragging a header. So I think it would make sense to have a nullable property containing all these drag-related properties

Reply:
done